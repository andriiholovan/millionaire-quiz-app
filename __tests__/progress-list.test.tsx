import {
  afterEach,
  beforeEach,
  describe,
  expect,
  it,
} from 'vitest';
import {
  cleanup,
  render,
  RenderResult,
  screen,
} from '@testing-library/react';
import ProgressList from '../app/_components/progress-list';

describe('PROGRESS LIST', () => {
  let container: RenderResult;
  const paramsId = 1;
  const quizList = [
    { step: 1, reward: '500' },
    { step: 2, reward: '64,000' },
    { step: 3, reward: '1,000,000' },
  ];

  beforeEach(() => {
    container = render(<ProgressList id={paramsId} quizList={quizList} />);
  });

  afterEach(cleanup);

  it('Should match snapshot', () => {
    expect(container.asFragment()).toMatchSnapshot();
  });

  it('Should render list of progress items', () => {
    const listItems = screen.getAllByRole('listitem');
    expect(listItems).toHaveLength(3);
  });
});
