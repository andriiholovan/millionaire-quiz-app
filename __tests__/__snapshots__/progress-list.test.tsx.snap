// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PROGRESS LIST > Should match snapshot 1`] = `
<DocumentFragment>
  <ul
    class="_progress_group_da84f6"
  >
    <li
      class="_progress_da84f6 _progress_current_da84f6"
    >
      500
    </li>
    <li
      class="_progress_da84f6 _progress_next_da84f6"
    >
      64,000
    </li>
    <li
      class="_progress_da84f6 _progress_next_da84f6"
    >
      1,000,000
    </li>
  </ul>
</DocumentFragment>
`;
